packages/react/react.ts

```diff
@@ -48,13 +50,17 @@ const isValidElement = (vnode?: VNode | VNode[] | null) => {
   return false;
 };
 
-const memo = (
-  component: (...args: unknown[]) => VNode
-) => {
+const memo = (component: (...args: unknown[]) => VNode) => {
+  let prevProps;
+  let prevRet;
   return (props: VProps) => {
-    return component(Object.values(props));
+    if (prevProps === props) return prevRet;
```

Comment:
This won't work as intended.

Consider this:

```jsx
/** @jsxRuntime automatic */
/** @jsxImportSource million */

const Component = memo((props) => {
  return <></>;
});

<Component props1="Hello, Million.js!" />
```

The JSX above will be compiled into:

```
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "million/jsx-runtime";
const Component = memo((props)=>{
    return _jsx(_Fragment, {});
});

_jsx(Component, {
    props1: "Hello, Million.js!"
});
```

As you can see, the props (`{ props1: "Hello, Million.js!" }`) is a plain object. When re-rendered, `prevProps === props` will always return false.

Reply:
This is very helpful and thanks for being patient with me. Let me update the code and run it by you again